include/wil/resource.h

```diff
@@ -2528,6 +2528,23 @@ namespace wil
 
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM)
     typedef unique_any<PSID, decltype(&::FreeSid), ::FreeSid> unique_sid;
+    typedef unique_any_handle_null_only<decltype(&::DeleteBoundaryDescriptor), ::DeleteBoundaryDescriptor> unique_boundary_descriptor;
+
+    namespace details
+    {
+        // Define a templated type in order to satisfy overload resolution below
+        template <ULONG flags>
+        struct private_namespace_helpers

```

Comment:
I don't think this needs a templated type. Can't this work?

```cpp
template<ULONG flags>
static void ClosePrivateNamespaceHelper(HANDLE h)
{
    ::ClosePrivateNamespace(h, flags);
}
```

Reply:
I based my code on the `FreePoolWithTag` wrapper. I thought that maybe there's a reason for this kind of definition. I can change it as you suggest, it seems to be working too.
https://github.com/microsoft/wil/blob/209ff9c7e30e71bf6d1405557bf78544920f8157/include/wil/resource.h#L6564-L6577

Reply:
Done. I also added `WI_NOEXCEPT`. Not sure about other properties such as `inline`, the headers are not very consistent in this regard.